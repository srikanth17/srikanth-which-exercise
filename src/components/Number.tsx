import React from 'react';

interface NumberProps {
  isActive: boolean;
  number: number;
  handleSelected: (num: number) => void;
}

const Number = ({ isActive, number, handleSelected }: NumberProps) => (
  <label
    className={`number${isActive ? ' active' : ''}`}
    data-testid="number-label"
  >
    <input
      checked={isActive}
      name="number"
      type="checkbox"
      value={number}
      onChange={(evt: any) => {
        handleSelected(number);
      }}
    />
    {number}
  </label>
);

export default Number;
