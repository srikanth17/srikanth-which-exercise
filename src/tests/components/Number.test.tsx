import React from 'react';
import { shallow } from 'enzyme';
import Number from '../../components/Number';

const handleSelectedSpy = jest.fn();

test('should render Number correctly', () => {
  const wrapper = shallow(
    <Number isActive={false} number={10} handleSelected={handleSelectedSpy} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should select a number', () => {
  const wrapper = shallow(
    <Number isActive={false} number={10} handleSelected={handleSelectedSpy} />
  );
  wrapper.find('input').simulate('change', { target: { checked: true } });
  expect(handleSelectedSpy).toBeCalled();
});
