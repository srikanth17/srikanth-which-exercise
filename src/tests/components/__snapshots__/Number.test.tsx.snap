// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Number correctly 1`] = `
<label
  className="number"
  data-testid="number-label"
>
  <input
    checked={false}
    name="number"
    onChange={[Function]}
    type="checkbox"
    value={10}
  />
  10
</label>
`;
